.github/workflows/UnitTests-core.yml

```diff
@@ -54,9 +54,10 @@ jobs:
         env:
           NODE_OPTIONS: "--max_old_space_size=8192"
         uses: mikepenz/action-junit-report@v3
-        if: always() # always run even if the previous step fails
+        if: failure() #  run  only if the previous step fails
```

Comment:
Let's keep always for now or put it immediately after ` Run Unit Tests with Code Coverage`